### PR TITLE
Plugin Refactor Fixes

### PR DESCRIPTION
**Pull request by aballman**
There were some issues that slipped under the radar.
This PR should address the issues discovered

### CODE REVIEW

File: plugins/kubernetes/deployment.go [hunk 1]
```diff
@@ -259,6 +260,8 @@ func genPodTemplateSpec(e transistor.Event, podConfig SimplePodSpec, kind string
 
 func (x *Kubernetes) doDeploy(e transistor.Event) error {
 	log.Info(e.Payload)
+	spew.Dump(e)
```
Comment (amoskyler) [inline plugins/kubernetes/deployment.go]: please remove

File: plugins/kubernetes/deployment.go [hunk 2]
```diff
@@ -405,6 +408,8 @@ func (x *Kubernetes) doDeploy(e transistor.Event) error {
 	batchv1DepInterface := clientset.BatchV1()
 
 	// Validate we have some services to deploy
+	spew.Dump(reData)
```
Comment (amoskyler) [inline plugins/kubernetes/deployment.go]: remove these as well

File: plugins/slack/slack.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ type Slack struct {
 func init() {
 	transistor.RegisterPlugin("slack", func() transistor.Plugin {
 		return &Slack{}
-	}, plugins.NotificationExtension{})
+	}, plugins.NotificationExtension{}, plugins.ProjectExtension{})
```
Comment (amoskyler) [inline plugins/slack/slack.go]: Is this necessary? I think I may have missed the point of registering multiple extension types

Comment (aballman) [reply plugins/slack/slack.go]: It is for tests. Basically, you need to tell transistor how to marshal these structs. When running the application, the registration for `ProjectExtension` is handled elsewhere. Once they are isolated in the tests, the registration doesn't happen.

The tests rely on the plugin they are testing to do this registration.

**Pull request merged**